qiskit_ibm_runtime/qiskit_runtime_service.py

```diff
@@ -1332,37 +1334,45 @@ def jobs(
         return [self._decode_job(job) for job in job_responses]
 
     def delete_job(self, job_id: str) -> None:
-        """Delete a runtime job.
+        """(DEPRECATED) Delete a runtime job.
 
         Note that this operation cannot be reversed.
 
         Args:
             job_id: ID of the job to delete.
 
         Raises:
-            RuntimeJobNotFound: If the job doesn't exist.
-            IBMRuntimeError: If the request failed.
+            RuntimeJobNotFound: The job doesn't exist.
+            IBMRuntimeError: Method is not supported.
         """
+
+        warnings.warn(
+            "The delete_job() method is deprecated and will be removed in a future release. "
+            "The new IBM Quantum Platform does not support deleting jobs.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+
         try:
             self._active_api_client.job_delete(job_id)
         except RequestsApiError as ex:
             if ex.status_code == 404:
                 raise RuntimeJobNotFound(f"Job not found: {ex.message}") from None
-            raise IBMRuntimeError(f"Failed to delete job: {ex}") from None
+            raise IBMRuntimeError(
+                "Failed to delete job. deleting jobs is not supported on the "
+                f"new IBM Quantum Platform: {ex}"
+            ) from None
```

Comment:
I would keep the generic "failed to delete job" message until `ibm_quantum` is fully removed. It's a small detail but users might bet confused if the error using the old platform refers to the new platform.